tests/unit/s2n_stuffer_hex_test.c

```diff
@@ -16,100 +16,572 @@
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_read_hex(NULL, &blob), S2N_ERR_NULL);
 
-    /* Read those back, and expect the same results */
-    for (int8_t i = 0; i < 50; i++) {
-        uint8_t value = i * (0xff / 50);
-        EXPECT_SUCCESS(s2n_stuffer_read_uint8_hex(&stuffer, &u8));
-        EXPECT_EQUAL(u8, value);
-    }
-    EXPECT_FAILURE(s2n_stuffer_read_uint8_hex(&stuffer, &u8));
+        /* s2n_stuffer_write_hex */
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_write_hex(&stuffer, NULL), S2N_ERR_NULL);
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_write_hex(NULL, &blob), S2N_ERR_NULL);
 
-    /* Try to write 26 2-byte ints bytes */
-    EXPECT_SUCCESS(s2n_stuffer_wipe(&stuffer));
-    for (uint16_t i = 0; i < 25; i++) {
-        uint16_t value = i * (0xffff / 25);
-        EXPECT_SUCCESS(s2n_stuffer_write_uint16_hex(&stuffer, value));
-    }
-    EXPECT_FAILURE(s2n_stuffer_write_uint16_hex(&stuffer, 1));
+        /* s2n_stuffer_read_uint8_hex */
+        uint8_t value_u8 = 0;
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_read_uint8_hex(&stuffer, NULL), S2N_ERR_NULL);
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_read_uint8_hex(NULL, &value_u8), S2N_ERR_NULL);
 
-    /* Read those back, and expect the same results */
-    for (uint16_t i = 0; i < 25; i++) {
-        uint16_t value = i * (0xffff / 25);
-        EXPECT_SUCCESS(s2n_stuffer_read_uint16_hex(&stuffer, &u16));
-        EXPECT_EQUAL(value, u16);
-    }
-    EXPECT_FAILURE(s2n_stuffer_read_uint16_hex(&stuffer, &u16));
-
-    /* Try to write 13 4-byte ints bytes */
-    EXPECT_SUCCESS(s2n_stuffer_wipe(&stuffer));
-    EXPECT_SUCCESS(s2n_stuffer_init(&stuffer, &b));
-    for (uint32_t i = 0; i < 12; i++) {
-        uint32_t value = i * (0xffffffff / 12);
-        EXPECT_SUCCESS(s2n_stuffer_write_uint32_hex(&stuffer, value));
+        /* s2n_stuffer_write_uint8_hex */
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_write_uint8_hex(NULL, 0), S2N_ERR_NULL);
+
+        /* s2n_stuffer_read_uint16_hex */
+        uint16_t value_u16 = 0;
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_read_uint16_hex(&stuffer, NULL), S2N_ERR_NULL);
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_read_uint16_hex(NULL, &value_u16), S2N_ERR_NULL);
+
+        /* s2n_stuffer_write_uint16_hex */
+        EXPECT_ERROR_WITH_ERRNO(s2n_stuffer_write_uint16_hex(NULL, 0), S2N_ERR_NULL);
     }
-    EXPECT_FAILURE(s2n_stuffer_write_uint32_hex(&stuffer, 1));
 
-    /* Read those back, and expect the same results */
-    for (uint32_t i = 0; i < 12; i++) {
-        uint32_t value = i * (0xffffffff / 12);
-        EXPECT_SUCCESS(s2n_stuffer_read_uint32_hex(&stuffer, &u32));
-        EXPECT_EQUAL(value, u32);
+    /* Test hex with uint8 */
+    {
+        const size_t expected_size = 2;
+        struct {
+            const uint8_t num;
+            const char *hex;
+        } test_cases[] = {
+            /* Test first digit */
+            { .num = 0, .hex = "00" },
```

Comment:
Nit, is there a reason why you're switching formats with the rest of the testcases?
```suggestion
            { .num = 0x00, .hex = "00" },
```

Reply:
I wanted to make it very clear that it was just zero. I can change it though if you don't think that's helpful.